### ISSUE TITLE
Circular referencing not working

### ISSUE THREAD

**Issue by wfuertes** [swagger-api/swagger-ui#1646]
Follow my context, include response class, end-point and js where the code brokers, because of a undefined property $ref.

Related: kongchen/swagger-maven-plugin#224

```
public class NodeTree {

    private Long id;
    private String name;
    private List<NodeTree> children;

    // GETTERS AND SETTERS
}
```

My documentated REST end-point.

```
@POST
@Path("/{treeId}/nodes")
@Produces({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
@Consumes({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
@ApiOperation(value = "Create node tree.")
@ApiImplicitParams(value = {
    @ApiImplicitParam(name = "nodeTree", value = "", required = true, dataType = "NodeTree",
                      paramType = "data")})
@ApiResponses(value = { @ApiResponse(code = 200, message = "Succees", response = NodeTree.class)})
public Response createNodeTree(@ApiParam(value = "Tree id", required = true) @PathParam("treeId") Long treeId, NodeTree nodeTree) {

    // IMPLEMENTATION OMITED 
}
```

So the swagger-maven-plugin is generating the follow swagger.json:

```
"NodeTree" : {
  "type" : "object",
  "required" : [ "name" ],
  "properties" : {
    "id" : {
      "type" : "integer",
      "format" : "int64"
    },
    "name" : {
      "type" : "string"
    },
    "children" : {
      "type" : "array"
      /* Here missing the follow part: 
          "items" : {
            "$ref" : "#/definitions/NodeTree"
          }
      */
    }
  }
}
```

Because this the "swagger-client.js" is breaking in the follow function:

---

```
Resolver.prototype.resolveTo = function (property, objs) {

  // it is breaking here, because we don't have 
  // $ref in the property to the type = array.
  var ref = property.$ref;

  if(ref) {
    if(ref.indexOf('http') === 0) {
      if(Array.isArray(objs[ref])) {
        objs[ref].push({obj: property, resolveAs: '$ref'});
      }
      else {
        objs[ref] = [{obj: property, resolveAs: '$ref'}];
      }
    }
  }
  else if(property.type === 'array') {
    var items = property.items;
    this.resolveTo(items, objs);
  }
}; 
----------------
```

I hope this can be useful in order to solve the problem.
Regards.
Willian.

**Comment by RobertPaasche** [comment]
Is there any improvement?

**Comment by RobertPaasche** [comment]
The relevant code comes from swagger-js.

New bug created:
https://github.com/swagger-api/swagger-js/issues/888

**Comment by webron** [comment]
This should be resolved in 3.0. Please reopen otherwise.
